packages/example/src/screens/HeadlessCheckoutVaultScreen.tsx

```diff
@@ -278,13 +284,23 @@ export default HeadlessCheckoutVaultScreen = (props: any) => {
             color: 'black'
           }} onPress={() => setSelectedVaultedPaymentMethod(item)}
           >
-            {'••••' + item.paymentInstrumentData.last4Digits}
+            {getVaultedPaymentData(item)}
           </Text>
         })
       );
     }
   }
 
+  const getVaultedPaymentData = (item: any) => {
+      const last4Digits = item.paymentInstrumentData.last4Digits
+      if (last4Digits !== undefined) {
+        return '••••' + last4Digits
+      }
+      const email = item.paymentInstrumentData?.sessionData?.billingAddress?.email ?? ""
+      const paymentMethodType = item?.paymentMethodType ?? ""
+      return paymentMethodType + ": " + email;
+  }
+
```

Comment:
[nit]: we could switch over the `paymentMethodType` here.

Reply:
@FlaviuExtPrimer we are trying to display relevant data for saved payment method.

Reply:
Changes pushed.